### PR TITLE
Add 1.29 to release schedule for 1.29 release

### PR DESCRIPTION
**Pull request by katcosgrove**
This PR adds 1.29 to the releases schedule for the 1.29.0 release.

/milestone 1.29
/cc @jimangel @marosset @Priyankasaggu11929

### CODE REVIEW

**Comment by Priyankasaggu11929** [comment]
cc: @kubernetes/release-managers

**Comment by sftim** [comment]
> It's a bit odd that we then use past tense for a future date

This is a preview of what people will see post-release.
When we release Kubernetes 1.29, the release will be in the past.

When reviewing, imagine that the release has already happened.

File: data/releases/schedule.yaml [hunk 1]
```diff
@@ -2,6 +2,14 @@
...
 # latest patch version for a minor release.
 schedules:
+- release: 1.29
+  releaseDate: 2023-12-13
+  next:
+    release: 1.29.1
+    cherryPickDeadline: 2024-01-12
+    targetDate: 2024-01-17
+  maintenanceModeStartDate: 2025-12-28
+  endOfLifeDate: 2026-02-28
```
Comment (xmudrii) [inline data/releases/schedule.yaml:12]: These dates are off for one year
```suggestion
  maintenanceModeStartDate: 2024-12-28
  endOfLifeDate: 2025-02-28
```

**Review by xmudrii — CHANGES_REQUESTED** [review]

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**